activitysim/abm/models/vehicle_allocation.py

```diff
@@ -0,0 +1,219 @@
+
+
+logger = logging.getLogger(__name__)
+
+
+def annotate_vehicle_allocation(model_settings, trace_label):
+    # - annotate tours table
+    tours = inject.get_table('tours').to_frame()
+    expressions.assign_columns(
+        df=tours,
+        model_settings=model_settings.get('annotate_tours'),
+        trace_label=tracing.extend_trace_label(trace_label, 'annotate_tours'))
+    pipeline.replace_table("tours", tours)
+
+
+def get_skim_dict(network_los, choosers):
+    skim_dict = network_los.get_default_skim_dict()
+    orig_col_name = 'home_zone_id'
+    dest_col_name = 'destination'
+
+    out_time_col_name = 'start'
+    in_time_col_name = 'end'
+    odt_skim_stack_wrapper = skim_dict.wrap_3d(orig_key=orig_col_name, dest_key=dest_col_name,
+                                               dim3_key='out_period')
+    dot_skim_stack_wrapper = skim_dict.wrap_3d(orig_key=dest_col_name, dest_key=orig_col_name,
+                                               dim3_key='in_period')
+
+    choosers['in_period'] = network_los.skim_time_period_label(choosers[in_time_col_name])
+    choosers['out_period'] = network_los.skim_time_period_label(choosers[out_time_col_name])
+
+    skims = {
+        "odt_skims": odt_skim_stack_wrapper.set_df(choosers),
+        "dot_skims": dot_skim_stack_wrapper.set_df(choosers),
+    }
+    return skims
+
+
+@inject.step()
+def vehicle_allocation(
+        persons,
+        households,
+        vehicles,
+        tours,
+        tours_merged,
+        network_los,
+        chunk_size,
+        trace_hh_id):
+    """Selects a vehicle for each occupancy level for each tour.
+
+    Alternatives consist of the up to the number of household vehicles plus one
+    option for non-household vehicles.
+
+    The model will be run once for each tour occupancy defined in the model yaml.
+    Output tour table will columns added for each occupancy level.
+
+    The user may also augment the `tours` tables with new vehicle
+    type-based fields specified via the annotate_tours option.
+
+    Parameters
+    ----------
+    persons : orca.DataFrameWrapper
+    households : orca.DataFrameWrapper
+    vehicles : orca.DataFrameWrapper
+    vehicles_merged : orca.DataFrameWrapper
+    tours : orca.DataFrameWrapper
+    tours_merged : orca.DataFrameWrapper
+    chunk_size : orca.injectable
+    trace_hh_id : orca.injectable
+
+    Returns
+    -------
+
+    """
+    trace_label = 'vehicle_allocation'
+    model_settings_file_name = 'vehicle_allocation.yaml'
+    model_settings = config.read_model_settings(model_settings_file_name)
+
+    logsum_column_name = model_settings.get('MODE_CHOICE_LOGSUM_COLUMN_NAME')
+
+    estimator = estimation.manager.begin_estimation('vehicle_allocation')
+
+    model_spec = simulate.read_model_spec(file_name=model_settings['SPEC'])
+    coefficients_df = simulate.read_model_coefficients(model_settings)
+    model_spec = simulate.eval_coefficients(model_spec, coefficients_df, estimator)
```

Comment:
if these are two different things, they should be two different names...

Reply:
changed.